### ISSUE TITLE
Singularity copies most host environment variables into container

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/galaxy#14408]
**Describe the bug**
Singularity copies most host environment variables into the container.
I am unsure if this had always been the case, but it doesn't seem like a good idea.
There is the `--cleanenv` option that prevents this behavior, we should use it.
That is closer to how docker works.

**Galaxy Version and/or server at which you observed the bug**
22.05, but should affect all releases

**To Reproduce**
1. Run a tool that prints the environment, and see that it also includes host variables

**Expected behavior**
Only explicitly set environment variables should be passed into the container

**Additional context**
Might be part of the problem that caused https://github.com/galaxyproject/tools-iuc/pull/4698

**Comment by mvdbeek** [comment]
Fixed in https://github.com/galaxyproject/galaxy/pull/14429

**Comment by microfuge** [comment]
Hi,

I pulled this update. The option for `cleanenv` is quite nice.  Overall, the containers run much better with our infrastructure than conda. 

The command constructed by our instance seems to be - 

`singularity --cleanenv  -s exec` which for our singularity version `v3.7.4` gives

```
Error for command "exec": unknown flag: --cleanenv
Usage:
  singularity [global options...] exec [exec options...] <container> <command>
```
Switching the command order worked -
```
singularity -s exec --cleanenv 
```
However, I tested only manually on terminal and disabling the `--cleanenv` with setting `singularity_cleanenv: false` in `job_conf.yml`  made the jobs run again. The `singularity_cmd` in job_conf was just the full path to singularity binary.
